### ISSUE TITLE
Undefined symbols for architecture arm64 for Cocoapods 1.6.0

### ISSUE THREAD

**Issue by bensmiley** [CocoaPods/CocoaPods#8502]
* [ ] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

Run `pod install` 

## What did you expect to happen?

I expected the library to compile correctly. It did with CocoaPods version 1.5.3.

## What happened instead?

```
Undefined symbols for architecture arm64:
  "_OBJC_CLASS_$_NSManagedObjectContext", referenced from:
      l_OBJC_$_CATEGORY_NSManagedObjectContext_$_RXPromise in RXPromise.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

### Stack

```
   CocoaPods : 1.5.3
        Ruby : ruby 2.3.7p456 (2018-03-28 revision 63024) [universal.x86_64-darwin18]
    RubyGems : 2.5.2.3
        Host : Mac OS X 10.14.2 (18C54)
       Xcode : 10.1 (10B61)
         Git : git version 2.10.1
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ ad728770f87f51d5ee72770d05be7d9d8d50745f
               master_old - https://github.com/CocoaPods/Specs.git @ 6ae3235a0399183fd63e936de972a4ebf2123c89
               private - https://bensmiley@bitbucket.org/bensmiley/pods.git @ 4f14e0b26b6c6e249d49c8fa37d0eba40dde3d99
```

### Installation Source

```
Executable Path: /usr/local/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.2
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.3.1
cocoapods-try         : 1.1.0
```

### Podfile

```ruby
source 'https://github.com/CocoaPods/Specs.git'
inhibit_all_warnings!
use_frameworks!

target 'ChatSDK Demo' do
  
  pod "ChatSDK", :path => "../"
  pod "ChatSDKFirebase/Adapter", :path => "../"
  pod "ChatSDKFirebase/FileStorage", :path => "../"
  pod "ChatSDKFirebase/Push", :path => "../"
  pod "ChatSDKFirebase/SocialLogin", :path => "../"
#   pod "ChatSDK/Extras", :path => "../"
  
#   pod 'TwitterKit', '<3.3.0'
#   pod 'TwitterCore', '<3.1.0'
#   pod 'GoogleSignIn'
#   pod 'FBSDKLoginKit'
 
  
  #pod "ChatSDKModules/StickerMessage", :path => "../ChatSDKModules"
  #pod "ChatSDKModules/KeyboardOverlayOptions", :path => "../ChatSDKModules"
  #pod "ChatSDKModules/ContactBook", :path => "../ChatSDKModules"

  #### Optional ####

  # Crashlytics see: https://firebase.google.com/docs/crashlytics/get-started?authuser=0

#   pod 'Fabric', '~> 1.7.6'
  pod 'Crashlytics', '~> 3.10.1'
  
end
```

## Project that demonstrates the issue

[https://github.com/chat-sdk/chat-sdk-ios](https://github.com/chat-sdk/chat-sdk-ios)

**Comment by amorde** [comment]
The link you posted doesn't seem to have a Podfile that matches what you posted - could you post an example project with the setup that you are testing? I tried the project in `Example` but that looks different than the Podfile you posted.

I did see that the example fails to build on 1.6.0 and succeeds on 1.5.3, but the error is something else.

**Comment by bensmiley** [comment]
@amorde I've updated the podfile in the description. This is the correct project to test: 

[https://github.com/chat-sdk/chat-sdk-ios/tree/master/Xcode](https://github.com/chat-sdk/chat-sdk-ios/tree/master/Xcode)

Steps to reproduce:

1. Install Cocoapods 1.5.3
2. Clone the example project
3. In the `Xcode` directory run `pod install`
4. Open the Xcode project and compile - it works
5. Install Cocoapods 1.6.0
6. Repeat steps 2 - 4
7. The error copied above is seen

I have gone through this exact process just now to confirm.

**Comment by chaoskyme** [comment]
me too, I revert 1.5.3.

**Comment by aminbenarieb** [comment]
Same here, @bensmiley I've managed to fix this my uninstalling Cocoapods 1.5.3.:
```
gem uninstall cocoapods --version 1.5.3
```
Also, try to clean the build folder and also do `pod install` from scratch.

In my case, I wanted to do a hotfix for my old branch, where all pods were installed with version 1.5.3, but my current version was 1.6.0. During the `pod install` process, I was getting a warning about version mismatch and after building my app crashed in runtime with message `image not found` of some pod dependency. So, I uninstalled 1.6.0, installed pods with 1.5.3 and everything worked fine. 

Then I returned to my current branch, installed 1.6.0, did `pod install` and got `Undefined symbols for architecture` after compilation.

**Comment by thisfin** [comment]
me too, I revert 1.6.0.beta.2

**Comment by tarun885** [comment]
Having same issue. Is there any fix yet?

**Comment by amorde** [comment]
I haven’t gotten the chance to take a deeper look yet

**Comment by amorde** [comment]
Okay I am able to reproduce - looking into the file that fails to build, it looks like it's related to RXPromise's use of the `weak_framework` attribute which isn't working correctly in 1.6.0 (see #8493)

I tested the fix for that issue and it works. This will be fixed in 1.6.1 (no promises on exact release timing) but for now if you'll need to stick to 1.5.3

Going to close this as it is already tracked. Thank you for the detailed bug report!

**Comment by sch1111878** [comment]
1.6.1 Still has this problem.
